extensions/common/vault/vault-aws/src/test/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *  Copyright (c) 2023 Amazon Web Services
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amazon Web Services - Initial Implementation
+ *
+ */
+
+package org.eclipse.edc.vault.aws;
+
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.CreateSecretRequest;
+import software.amazon.awssdk.services.secretsmanager.model.DeleteSecretRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+
+class AwsSecretsManagerVaultTest {
+
+    private final Monitor monitor = Mockito.mock(Monitor.class);
+    private final SecretsManagerClient secretClient = Mockito.mock(SecretsManagerClient.class);
+
+    private final AwsSecretsManagerVault vault = new AwsSecretsManagerVault(secretClient, monitor);
+
+    @Test
+    void resolveSecret_sanitizeKeyNameReplacesInvalidCharacters() {
+        for (var validCharacter : List.of('_', '+', '-', '@', '/', '.')) {
+            var validKey = "valid" + validCharacter + "key";
+            assertThat(vault.sanitizeKey(validKey)).isEqualTo(validKey + '_' + validKey.hashCode());
+        }
+        var key2 = "invalid#key";
+        var sanitized = vault.sanitizeKey(key2);
+        assertThat(sanitized).isEqualTo("invalid-key" + "_" + key2.hashCode());
+    }
+
+    @Test
+    void resolveSecret_sanitizeKeyNameLimitsKeySize() {
```

Comment:
please use empty lines to separate "arrange" "act" and "assert" phases of the test